### PR TITLE
test(date): address flakiness

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Adds a new helper function `toBeFocusedDelayed` and utilises the current `containsClass` helper to address frequently flaking tests 

### Current behaviour

Tests in date are flaking

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: playwright/support/helper.ts [hunk 1]
```diff
@@ -193,6 +193,17 @@ export const containsClass = async (
...
 };
 
+/**
+ * Uses the .toHaveFocus() assertion with a waitFor before to
+ * ensure the locator is available, and enough time has passed for the focus to be set.
+ * @param locatorFunc the locator you'd like to use
+ * @example await toBeFocusDelayed(exampleLocator(page));
+ */
+export const toBeFocusedDelayed = async (locatorFunc: Locator) => {
+  await locatorFunc.waitFor({ timeout: 10000 });
```
Comment (damienrobson-sage) [inline playwright/support/helper.ts:203]: **question**: Purely for my own knowledge, why 10 seconds? Why not e.g. 5? Just wondering whether it came from your investigation or whether it's just a number plucked from the aether.

**Review by damienrobson-sage — DISMISSED** [review]
LGTM 👍

File: playwright/support/helper.ts (hunk 1 above)
Comment (tomdavies73) [reply playwright/support/helper.ts:203]: I believe 5 is the standard, so double that just to account for extra flakiness. Depending on what the second reviewer thinks, I'm happy to let this be higher or lower though 👍

File: playwright/support/helper.ts [hunk 2]
```diff
@@ -193,6 +193,17 @@ export const containsClass = async (
   await expect(locatorFunc).toHaveClass(classNameRegEx);
 };
 
+/**
+ * Uses the .toHaveFocus() assertion with a waitFor before to
+ * ensure the locator is available, and enough time has passed for the focus to be set.
+ * @param locatorFunc the locator you'd like to use
+ * @example await toBeFocusDelayed(exampleLocator(page));
```
Comment (DipperTheDan) [inline playwright/support/helper.ts:200]: **nitpick:** The nittiest of nitpicks.

```suggestion
 * @example await toBeFocusedDelayed(exampleLocator(page));
```

**Review by DipperTheDan — APPROVED** [review]

**Review by damienrobson-sage — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**